### ISSUE TITLE
Bug/notification-email enabling causes crowdsec/main to crash 

### ISSUE THREAD

**Issue by jpalko** [crowdsecurity/crowdsec#1180]
**Describe the bug**
I am trying to enable notification-email and once I have modified /etc/crowdsec/profiles.yaml and /etc/crowdsec/notifications/email.yaml and try to restart the service systemd tells that it does not start.

**To Reproduce**
1. Configure /etc/crowdsecnotifications/email.yaml for google mail delivery
2. Uncomment in profiles.yaml:
```
notifications:
 - email_default
```
3. systemd restart crowdsec
4. See error:
```
Job for crowdsec.service failed because the control process exited with error code.
See "systemctl status crowdsec.service" and "journalctl -xeu crowdsec.service" for details.
```

**Expected behavior**
Service starts and email notifications get enabled.

**Technical Information (please complete the following information):**
 - OS: Fedora
 - Version 35
 - SELinux targeted policy enabled

**Additional context**
/var/log/crowdsec.log:
```
time="18-01-2022 18:55:37" level=info msg="Crowdsec v1.2.3-linux-a6e405422c732e9b6c46ae1004a2b80297df8336"
time="18-01-2022 18:55:37" level=info msg="Loading prometheus collectors"
time="18-01-2022 18:55:37" level=info msg="Loading CAPI pusher"
time="18-01-2022 18:55:37" level=info msg="initiating plugin broker"
time="18-01-2022 18:55:37" level=error msg="crowdsec - goroutine crowdsec/main crashed : runtime error: invalid memory address or nil pointer dereference"
time="18-01-2022 18:55:37" level=error msg="please report this error to https://github.com/crowdsecurity/crowdsec/"
time="18-01-2022 18:55:37" level=error msg="stacktrace/report is written to /tmp/crowdsec-crash.349495449.txt : please join it to your issue"
time="18-01-2022 18:55:37" level=fatal msg="crowdsec stopped"
```

/tmp/crowdsec-crash.349495449.txt:
```
error : runtime error: invalid memory address or nil pointer dereference
version: v1.2.3-a6e405422c732e9b6c46ae1004a2b80297df8336
Codename: alphaga
BuildDate: 2022-01-06_10:23:53
GoVersion: 1.17.5
Platform: linux
goroutine 1 [running]:
runtime/debug.Stack()
        /opt/hostedtoolcache/go/1.17.5/x64/src/runtime/debug/stack.go:24 +0x65
github.com/crowdsecurity/crowdsec/pkg/types.WriteStackTrace({0x11bc2c0, 0x1d95a50})
        /home/runner/work/crowdsec/crowdsec/pkg/types/utils.go:104 +0x254
github.com/crowdsecurity/crowdsec/pkg/types.CatchPanic({0x1326ef7, 0xd})
        /home/runner/work/crowdsec/crowdsec/pkg/types/utils.go:119 +0xfa
panic({0x11bc2c0, 0x1d95a50})
        /opt/hostedtoolcache/go/1.17.5/x64/src/runtime/panic.go:1038 +0x215
github.com/crowdsecurity/crowdsec/pkg/csplugin.(*PluginBroker).loadNotificationPlugin(0x1e275e0, {0xc0004affad, 0x5}, {0xc0004aff80, 0x32})
        /home/runner/work/crowdsec/crowdsec/pkg/csplugin/broker.go:244 +0x125
github.com/crowdsecurity/crowdsec/pkg/csplugin.(*PluginBroker).loadPlugins(0x1e275e0, {0xc0004ab900, 0x0})
        /home/runner/work/crowdsec/crowdsec/pkg/csplugin/broker.go:212 +0x18c
github.com/crowdsecurity/crowdsec/pkg/csplugin.(*PluginBroker).Init(0x1e275e0, 0xc0000ac6e0, {0xc0000aa2e8, 0x1, 0x1}, 0xc0004e6930)
        /home/runner/work/crowdsec/crowdsec/pkg/csplugin/broker.go:87 +0x1e5
main.initAPIServer(0xc0004e4000)
        /home/runner/work/crowdsec/crowdsec/cmd/crowdsec/api.go:30 +0x131
main.Serve(0xc0004e4000)
        /home/runner/work/crowdsec/crowdsec/cmd/crowdsec/serve.go:226 +0x25e
main.main()
        /home/runner/work/crowdsec/crowdsec/cmd/crowdsec/main.go:310 +0x3e6
```

**Comment by buixor** [comment]
Hello @jpalko !

Can you provide your email.yaml conf (without the sensitive info ofc) ?
It might be related to SELinux, but it's just a wild guess.

**Comment by jpalko** [comment]
I tried also setting SELinux permissive mode on and that did not alter behaviour.

**Comment by buixor** [comment]
Hello @jpalko !

You didn't use the package to deploy, did you ?

So there are few things :
 - a crash in the code (being fixed by #1181) that happened if we set the plugin to run with non-existing group or user
 - the fact that centos&co doesn't have the `nogroup` group (and the package takes care to replace with nobody) which caused the crash on your machine
 - ... the fact that we didn't yet distribute package for fedora35, which we will do in the next release.

In the meanwhile, can you try to set the group to `nobody` in the `plugin_conf` of config.yaml of crowdsec ?

I keep this issue open, I'll close as soon as you confirm this solves and we add the fedora 35 build target :)

**Comment by jpalko** [comment]
Hello @buixor

Yes, I used the tarball install as the Fedora packaging is only available up to version 34 and that one has two issues:

- It requests the package yum-utils which does get resolved by package management to dnf-utils
- It requests the package pygpgme whereas the package has been renamed to gpgme

I edited config.yaml to read:
```
plugin_config:
  user: nobody # plugin process would be ran on behalf of this user
  group: nobody # plugin process would be ran on behalf of this group
```
After this crowdsec is looking much better:
```
# systemctl status crowdsec
● crowdsec.service - Crowdsec agent
     Loaded: loaded (/etc/systemd/system/crowdsec.service; enabled; vendor preset: disabled)
     Active: active (running) since Wed 2022-01-19 12:52:21 EET; 6s ago
    Process: 1335 ExecStartPre=/usr/local/bin/crowdsec -c /etc/crowdsec/config.yaml -t (code=exited, status=0/SUCCESS)
   Main PID: 1350 (crowdsec)
      Tasks: 20 (limit: 9477)
     Memory: 33.8M
        CPU: 2.075s
     CGroup: /system.slice/crowdsec.service
             ├─1350 /usr/local/bin/crowdsec -c /etc/crowdsec/config.yaml
             └─1357 /usr/local/lib/crowdsec/plugins/notification-email
```

Thanks for the quick responses, my install should be working nicely now. :)
